File: app/src/main/java/us/logaming/myufrplanning/ui/overview/OverviewViewModel.java
package us.logaming.myufrplanning.ui.overview;

import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;
import java.util.concurrent.Executor;

import us.logaming.myufrplanning.data.repositories.PlanningRepository;
import us.logaming.myufrplanning.data.Result;
import us.logaming.myufrplanning.data.model.PlanningItem;

public class OverviewViewModel extends ViewModel {
    private final MutableLiveData<List<PlanningItem>> planningItems;
    private final PlanningRepository planningRepository;

    public OverviewViewModel(Context context, Executor executor) {
        this.planningItems = new MutableLiveData<>(null);
        this.planningRepository = new PlanningRepository(context, executor);
    }

    public LiveData<List<PlanningItem>> getPlanningItems() {
        loadPlanningItems();
        return this.planningItems;
    }

    private void loadPlanningItems() {
        this.planningRepository.fetchLatestPlanning(result -> {
            if (result instanceof Result.Success) {
               this.planningItems.postValue(((Result.Success<List<PlanningItem>>) result).data);
            } else {
                this.planningItems.postValue(null);
            }
        }, true);
    }
}

File: app/src/main/java/fr/loganh/myufrplanning/ui/overview/OverviewViewModel.kt
package fr.loganh.myufrplanning.ui.overview

import androidx.lifecycle.*
import dagger.hilt.android.lifecycle.HiltViewModel
import fr.loganh.myufrplanning.data.repository.PlanningRepository
import fr.loganh.myufrplanning.model.PlanningItem
import kotlinx.coroutines.launch
import javax.inject.Inject

enum class SednaApiStatus {
    LOADING,
    ERROR,
    DONE
}

@HiltViewModel
class OverviewViewModel @Inject constructor(
    private val planningRepository: PlanningRepository
) : ViewModel() {

    // The internal MutableLiveData String that stores the most recent request
    private val _status = MutableLiveData<SednaApiStatus>()

    // The external immutable LiveData for the request status
    val status: LiveData<SednaApiStatus> = _status

    private val _planning = MutableLiveData<List<PlanningItem>>()

    val planning: LiveData<List<PlanningItem>> = _planning

    init {
        getPlanning(listOf(5630), 14)
    }

    private fun getPlanning(groupIds: List<Int>, nbDays: Int) {
        viewModelScope.launch {
            _status.value = SednaApiStatus.LOADING

            try {
                _planning.value = planningRepository.fetchPlanning(groupIds, nbDays)
                _status.value = SednaApiStatus.DONE
            }
            catch (e: Exception) {
                _status.value = SednaApiStatus.ERROR
                _planning.value = emptyList()
            }
        }
    }
}